packages/server/src/manifest-generator.ts

```diff
@@ -39,37 +36,48 @@ module.exports = {
   children: children,
 }
 `
-
   yield mkdir(path.dirname(options.destinationPath), { recursive: true });
   yield writeFile(options.destinationPath, manifest);
 }
 
-export function* createManifestGenerator(options: ManifestGeneratorOptions): Operation {
-  if(options.watch) {
-    let watcher = chokidar.watch(options.files, { ignoreInitial: true });
+export const manifestGenerator: Service<ManifestGeneratorOptions> = function *(options) {
+  let { atom, files, mode } = options;
 
-    yield ensure(() => watcher.close());
+  assert(atom, 'No atom in manifestGenerator options');
+
+  let serviceStatus = atom.slice('manifestGenerator', 'status');
+  
+  serviceStatus.set({ type: 'started' });
 
-    let events: Mailbox = yield Mailbox.subscribe(watcher, ['ready', 'add', 'unlink']);
+  if(mode === 'watch') {
+    let watcher = chokidar.watch(files, { ignoreInitial: true });
+
+    yield ensure(() => watcher.close());
 
     yield throwOnErrorEvent(watcher);
 
-    yield events.receive({ event: 'ready' });
+    yield once(watcher, 'ready');
     yield writeManifest(options);
 
-    console.debug("[manifest generator] manifest ready, watching for updates");
-    options.delegate.send({ status: 'ready' });
+    console.debug("[manifest generator] manifest ready, watching for updates")
 
-    while(true) {
-      yield events.receive();
-      yield writeManifest(options);
+    serviceStatus.update(() => ({ type: 'reachable' }));
+    
+    let fileChanges = new Channel<void>();
 
+    let writeOperation = function *() {
+      fileChanges.send();
       console.debug("[manifest generator] manifest updated");
-      options.delegate.send({ event: 'update' });
+      serviceStatus.update(() => ({ type: 'reachable' }))
     }
+
+    yield spawn(on(watcher, 'add').forEach(writeOperation));
+    yield spawn(on(watcher, 'unlink').forEach(writeOperation));
+    
+    yield subscribe(fileChanges).forEach(() => writeManifest(options));
   } else {
     yield writeManifest(options);
     console.debug("[manifest generator] manifest ready");
```

Comment:
Is this now redundant since our sync-block will print this out?

Reply:
@cowboyd the eventual goal is for these `console.debug` statements just implicityly happen.  

What is the `sync-block` you mention here?

Reply:
I mean the big pile 'o fork that waits for all services to be ready in `orchestrator.ts`

Reply:
oh, you mean the cutlery.

`console.debug` removed